### PR TITLE
fix: fix browser build for Webpack 5

### PR DESCRIPTION
**Pull request by vkarpov15**
Re: #11584

**Summary**

With #11584, there were a few issues that popped up in the browser build:

1. Missing `Buffer`
2. Missing `process`
3. Looks like Babel was failing on `isAsyncFunction()`

**Examples**

### CODE REVIEW

**Review by AbdelrahmanHafez — APPROVED** [review]
LGTM, thanks! 👍

File: lib/helpers/isAsyncFunction.js [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-const asyncFunctionPrototype = Object.getPrototypeOf(async function() {});
+let asyncFunctionPrototype = null;
+// try/catch for Babel compatibility, because Babel preset-env requires
+// regenerator-runtime for async/await and we don't want to include that
+// for a simple check.
+try {
+  asyncFunctionPrototype = Object.getPrototypeOf(async function() {});
+} catch (err) {}
 
 module.exports = function isAsyncFunction(v) {
```
Comment (Uzlopak) [inline lib/helpers/isAsyncFunction.js:11]: we could avoid a potential performance penalty by checking the value of asyncFunctionPrototype only once.

```javascript
module.exports =  asyncFunctionPrototype !== null
? function isAsyncFunction(v) {
  return (
    typeof v === 'function' &&
    Object.getPrototypeOf(v) === asyncFunctionPrototype
  );
}
: function () { return false; };
```

**Comment by Uzlopak** [comment]
What is the actual purpose of the browser build?

**Comment by vkarpov15** [comment]
@Uzlopak the browser build is supposed to let you run basic schema validation in the browser. The syntax is slightly different: there's no models, you need to define a schema when you create a document. But you can at least define and use Mongoose schemas in the browser. For example:

```html
<html>
  <head>
    <script src="/dist/browser.umd.js"></script>
    <script>
      const schema = new mongoose.Schema({
        name: { type: String, required: true },
        quest: { type: String, match: /Holy Grail/i, required: true },
        favoriteColor: { type: String, enum: ['Red', 'Blue'], required: true }
      });

      doc = new mongoose.Document({
        name: 'Sir Lancelot of Camelot',
        quest: 'To seek the Holy Grail',
        favoriteColor: 'Blue'
      }, schema);
      console.log(doc.validateSync()); // null

      doc.set({
        name: 'Sir Galahad of Camelot',
        quest: 'To seek the Holy Grail',
        favoriteColor: 'Yellow'
      });

      console.log(doc.validateSync()); // ValidationError
    </script>
  </head>
  <body></body>
</html>
```

**Comment by AbdelrahmanHafez** [comment]
I'm genuinely curious if anyone out there is relying on that.

**Pull request merged**